packages/@react-aria/checkbox/src/useCheckbox.ts

```diff
@@ -61,6 +62,18 @@ export function useCheckbox(props: AriaCheckboxProps, state: ToggleState, inputR
     }
   });
 
+  // Reset validation state on label click for checkbox with a hidden input.
+  state.toggle = chain(state.toggle, () => {
```

Comment:
A coworker pointed out that this is mutating the state object and thus if the child that called `useCheckbox` is rerendering more than the state is being recreated then multiple `state.toggles` could be chained here. What do you think about just calling `usePress` in here and duplicating what `useToggle` does for labelProps? A bit gross that we'd be duplicating the code but maybe good enough for now until we decide whether or not useToggle should accept press handlers.

Reply:
Good catch 👍 

I don't think it matters as `useToggleState` isn't memoized anyways, but we can `usePress` here if you want to make sure.

Reply:
I still don't like mutating the state object, it breaks some assumptions and makes it harder to debug.
I'd prefer to just duplicate the code.